src/state/atoms/releaseAtom.ts

```diff
@@ -1,5 +1,22 @@
+import axios from 'axios';
+import { updateVisibilityFunctionsBeta, visibilityFunctionsExist } from '../../utils/VisibilitySingleton';
 import { atomWithToggle } from './utils';
+import { getUnleashClient, unleashClientExists } from '../../components/FeatureFlags/unleashClient';
 
-import { isBeta } from '../../utils/common';
-
-export const isPreviewAtom = atomWithToggle(isBeta());
+export const isPreviewAtom = atomWithToggle(undefined, (isPreview) => {
+  try {
+    // Required to change the `isBeta` function return value in the visibility functions
+    if (visibilityFunctionsExist()) {
+      updateVisibilityFunctionsBeta(isPreview);
+      axios.post('/api/chrome-service/v1/user/update-ui-preview', { uiPreview: isPreview });
+    }
+    if (unleashClientExists()) {
+      // Required to change the `platform.chrome.ui.preview` context in the feature flags, TS is bugged
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      getUnleashClient().updateContext({ 'platform.chrome.ui.preview': isPreview });
+    }
+  } catch (error) {
+    console.error('Failed to update the visibility functions or feature flags context', error);
```

Comment:
@Hyperkid123 thanks for adding some error handling - however are we sure this actually catches the errors that can occur above? 

In codepen I tried this snippet of code with the axios library - and the catch never executes (even when a 404 occurs to the API being post-ed):

```js
try {
  console.info('Trying the post');
  axios.post('/foo/bar', { fooBar: 'yes' });
} catch(e) {
  console.warn('I was caught');
  console.warn(e);
}
```
Which results in:
![image](https://github.com/RedHatInsights/insights-chrome/assets/21317056/8b809c0a-0580-4315-b846-e1c799e5b116)

I think we may need to add the catch to the promise itself?

Reply:
I also confirmed it's not being hit in chrome by blocking the network request for update-preview chrome service API: 
![image](https://github.com/RedHatInsights/insights-chrome/assets/21317056/4029bdbb-1b85-47cb-8e30-ce94757fdeca)

Reply:
Magnificent. I though we only had to be paranoid about fetch not handling 4xx

Reply:
Oh its because the atom is not async itself. Easy fix.

Reply:
Verified this now works. Thanks!